File: Algorithms/Set Mismatch.py
class Solution:
    def findErrorNums(self, nums: List[int]) -> List[int]:
        n=len(nums)
        expected=(n*(n+1))//2
        actual=0
        d={}
        for i in range(n):
            if nums[i] in d:
                duplicate=nums[i]
            d[nums[i]]=None
            actual+=nums[i]
        return[duplicate,expected-actual+duplicate]

File: Algorithms(Java)/Set Mismatch.java
class Solution {
    public int[] findErrorNums(int[] nums) {
        int n=nums.length;
        int expected=(n*(n+1))/2;
        int actual=0;
        HashSet<Integer> h=new HashSet<Integer>();
        int duplicate=0;
        for (int i:nums){
            if (h.contains(i))
                duplicate=i;
            h.add(i);
            actual+=i;
        }
        return new int[] {duplicate,expected-actual+duplicate};
    }
}